File: PHP/data_structures/arrays/stack.php
<?php

/**
 * Implement a stack data structure with arrays
 */

class Stack {
    private $arr;

    public function __construct() {
        $this->arr = [];
    }

    public function push($data) {
        $this->arr[] = $data;
        return;
    }

    public function pop() {
        if ($this->isEmpty()) {
            echo "Stack is empty." . PHP_EOL;
            return;
        }
        return array_pop($this->arr);
    }

    public function top() {
        return end($this->arr);
    }

    public function isEmpty() {
        return empty($this->arr);
    }

    public function printStack() {
        echo implode(" ", $this->arr) . PHP_EOL;
    }
}

$stack = new Stack();
$stack->push(1);
$stack->push(2);
$stack->push(3);
$stack->push(4);
$stack->push(5);
$stack->printStack();
$stack->pop();
$stack->printStack();
$stack->pop();
$stack->printStack();
$stack->pop();
$stack->printStack();
$stack->pop();
$stack->printStack();
$stack->pop();
$stack->printStack();
$stack->pop();
$stack->printStack();

File: Python/stack_queues/stack.py
# implementing a stack


class Stack(object):
    """

    Implementing a stack using list data structure.

    """
    def __init__(self):
        self.stack = []

    def push(self, value):
        self.stack.append(value)

    def pop(self):
        if len(self.stack) < 1:
            return
        return self.stack.pop()

    def print_stack(self):
        print self.stack
        return


if __name__ == "__main__":
    stack = Stack()
    stack.push(1)
    stack.push(2)
    stack.push(3)
    stack.push(4)
    stack.print_stack()
    stack.pop()
    stack.print_stack()
    stack.pop()
    stack.pop()
    stack.pop()
    stack.print_stack()
